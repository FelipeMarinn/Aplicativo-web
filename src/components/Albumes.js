import React, { useEffect } from 'react'
import { useDispatch } from 'react-redux'
import { useSelector } from 'react-redux'
import styled from 'styled-components'
import { setData } from '../helpers/setData'

const AlbumesStyled = styled.div`
  max-width: 1000px;
  margin: 0 auto;
  margin-top: 100px;

  display: grid;
  grid-row-gap: 2.3em;
  grid-auto-flow: columns;
  grid-column-gap: 20px;
  grid-template-columns: repeat(auto-fill, minMax(0, 300px));
  background: var(--background);
  justify-content: center;

  .album-content {
    background-color: white;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
    margin-bottom: 25px;
  }

  .album-info {
    padding: 10px 20px;
  }
`

export const Albumes = () => {

    const dispatch = useDispatch()
    const { albumList } = useSelector(state => state.albumsReducer)

    useEffect(() => {

        setData('albums').then( data => dispatch({
          type: 'GET_DATA_ALBUMS',
          payload: data
        }))
    
      }, [dispatch])

    return (
        <AlbumesStyled>
            {
                albumList.map( ({ title, id }) => {
                    return (
                        <div key={ title } className='album-content'>
                            <img src='https://www.tuexpertoapps.com/wp-content/uploads/2017/10/old-1130731_640-600x450.jpg' alt='camara fotografica' width='100%'/>
                            <div className='album-info'>
                               <p className='album-id'><strong>Id:</strong> { id } </p>
                               <p><strong>Title:</strong> { title } </p>
                            </div>
                        </div>
                    )
                })
            }
        </AlbumesStyled>
    )
}
