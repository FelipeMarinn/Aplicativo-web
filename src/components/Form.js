import React, { useState } from 'react'
import { useDispatch } from 'react-redux'
import TextField from '@material-ui/core/TextField';
import styled from 'styled-components'

const FormStyled = styled.div`

  .inputPost {
      margin-right: 20px;
      height: 3em;
  }

  .btn {
    height: 56px;
    width: 100px;
    padding: 10px;
    border: none;
    border-radius: 4px; 
    font-size: 1em;
    color: #fff;
    background-color: #0781b2;
  }
  form {
    text-align: center;
  }

  @media screen and (max-width: 768px) {
    .inputPost {
      display: block;
      margin-right: 0;
      margin-bottom: 20px;
    }
    form {
      display: block;
      width: 60%;
      margin: 0 auto;
      text-align: left;
    }
  }
`


export const Form = () => {

    const dispatch = useDispatch()

    const [inputTitle, setInputTitle] = useState('')
    const [inputBody, setInputBody] = useState('')

    const [leyendaTitle, setLeyendaTitle] = useState('')
    const [leyendaBody, setLeyendaBody] = useState('')

    const [ errorTitle, setErrorTitle ] = useState(false)
    const [ errorBody, setErrorBody ] = useState(false)


    const handleInputBody = (e) => {
        const value = e.target.value
        setInputBody(value)

        if ( value.length < 10 || value.length > 100 ) {
            setErrorBody(true)
            setLeyendaBody('debe tener entre 10 a 100 caracteres')
        } else {
            setErrorBody(false)
            setLeyendaBody('')
        }
    }

    const handleInputTitle = (e) => {
        const value = e.target.value
        setInputTitle(value)

        if ( value.length < 5 ) {
            setErrorTitle(true)
            setLeyendaTitle('debe tener almenos 5 caracteres')
        } else {
            setErrorTitle(false)
            setLeyendaTitle('')
        }
    }

    const handleSubmit = (e) => {

        e.preventDefault()

        if ( inputTitle && inputBody ) {
            if ( !errorTitle && !errorBody ) {
                fetch('https://jsonplaceholder.typicode.com/posts', {
                    method: 'POST',
                    body: JSON.stringify({
                      title: inputTitle,
                      body: inputBody,
                      userId: 1,
                    }),
                    headers: {
                      'Content-type': 'application/json; charset=UTF-8',
                    },
            
                  })
                  .then((response) => response.json())
                  .then((json) => dispatch({
                      type: 'SET_POST_DATA',
                      payload: json
                  }))
                  
            } else {
                return false
            }
        }
      
    }

    return (
        <FormStyled>
            <form>
              <TextField
                className='inputPost'
                error={ errorTitle }
                label="Some awesome title"
                helperText={ leyendaTitle }
                variant="filled"
                value={ inputTitle }
                onChange={ handleInputTitle }
              />
              <TextField
                className='inputPost'
                error={ errorBody }
                label="What happened today"
                helperText={ leyendaBody }
                multiline
                variant="filled"
                value={ inputBody }
                onChange={ handleInputBody }
              />

              <button className='btn' type='submit' onClick={ handleSubmit } > Publicar </button>  
            </form>
        </FormStyled>
    )
}
