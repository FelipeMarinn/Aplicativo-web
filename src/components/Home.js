import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import styled from 'styled-components'
import { setData } from '../helpers/setData'

const HomeStyled = styled.div`
    max-width: 1000px;
    margin: 0 auto;
    margin-top: 100px;
    padding-bottom: 100px;

    display: grid;
    justify-content: center;
    aling-items: center;
    grid-gap: 10px;
    grid-template-columns:  repeat(auto-fill, minMax(0, 300px));
    grid-template-rows: 100px, 100px;
    grid-template-areas:
    "header  header  header" 
    "sidebar content content";
        
    .image-content-1 {
      height: 350px;
      width: 100%;
      grid-area: header;
    } 
    .image-content-2 {
      height: 350px;
      width: 100%;
      grid-area: content;
    } 
    .image-content-3 {
      height: 350px;
      width: 100%;
      grid-area: sidebar;
    }  

    @media screen and (max-width: 627px) { 
      grid-template-rows: repeat(1fr);
      grid-template-areas:
        "header  header  header" 
        "sidebar sidebar sidebar"
        "content content content";
    }      
 
`

export const Home = () => {

    const dispatch = useDispatch()
    const { photoList } = useSelector(state => state.homeReducer)

    useEffect(() => {

      setData('photos').then( data => dispatch({
        type: 'GET_DATA_PHOTOS',
        payload: data
      }))
  
    }, [dispatch])

    return (
        <HomeStyled>
            {
                photoList.map( ({ title, url, id }) => {
                    return (
                        <div 
                          key={ title } 
                          className={`image-content-${id}`}
                          style={{ backgroundImage: `url(${ url })` }}></div>
                    )
                })
            }
        </HomeStyled>   
    )
}
