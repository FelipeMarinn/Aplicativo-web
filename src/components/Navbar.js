import React, { useState } from 'react'
import styled from 'styled-components'
import { Link, NavLink } from 'react-router-dom'

const NavbarStyled = styled.div`
 
  padding: 10px 0;
  background-color: #f2f6fc;

  nav {
    width: 1000px;
    margin: 0 auto;
    height: 40px;
   
    .nav-item {
      color: black;
      text-align: center;
      padding: 0 15px;
      line-height: 40px;
      text-decoration: none;
      letter-spacing: 1px;
      &:hover {
          color: #0781b2;
      } 
    }
  }
  .bars {
      margin-left: 10px;
  }
  .bars, .close {
    display: none;  
    font-size: 20px;
  }

  @media screen and (max-width: 650px) {
      nav {
        position: fixed;
        top: -100%;
        left: 0;
        width: 100%;
        height: 100vh;
        display: flex;
        justify-content: space-around;
        align-items: center;
        flex-direction: column;
        background-color: white;
        color: black;
        z-index: 1000;
        opacity: 0.95;
        transition: 0.3s linear;
      }
    nav.toggle {
        top: 0;
    }

    .bars, .close {
        display: inline-block;  
    }
    .close{
        position: absolute;
        right: 50px;
        top: 30px;
    }
  }
`


export const Navbar = () => {

    const [toggle, setToggle] = useState(false)

    const menuToggle = () => {
        setToggle( !toggle )
    }

    return (
        <NavbarStyled>
            
            <div className='bars' onClick={ menuToggle }>
              <i className="fas fa-bars" ></i> 
            </div>
          
            <nav className={ toggle ? 'toggle' : '' }>
                
                <Link  to="/" className="nav-item" onClick={ menuToggle } >
                    Home
                </Link>
    
                <NavLink
                    activeClassName="active"
                    className="nav-item" 
                    exact
                    to="/usuarios"
                    onClick={ menuToggle }
                >
                    Usuarios
                </NavLink>       
                <NavLink 
                    activeClassName="active"
                    className="nav-item" 
                    exact
                    to="/albumes"
                    onClick={ menuToggle }
                >
                    Albumes
                </NavLink>       
                <NavLink 
                    activeClassName="active"
                    className="nav-item" 
                    exact
                    to="/publicaciones"
                    onClick={ menuToggle }
                >
                    publicaiones
                </NavLink>

                <div className='close'>
                  <i className="fas fa-times" onClick={ menuToggle }></i>    
                </div>
                
        </nav>
              
        </NavbarStyled>
    )
}
