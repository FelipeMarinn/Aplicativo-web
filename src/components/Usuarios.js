import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import styled from 'styled-components'

const UsuariosStyled = styled.div`
    max-width: 1000px;
    margin: 0 auto;
    margin-top: 100px;

    input {
        display: block;
        margin: 0 auto;
        margin-bottom: 100px;
        text-align: center;
        border: none;
        height: 48px;
        line-height: 48px;
        box-shadow: 0 0 5px rgba(0,0,0,.09);
        outline: 0;
        color: var(--black);
        background: var(--white);
        &::-webkit-input-placeholder {
            color: #c4c4c4;
        }
        &::-webkit-search-cancel-button {
          display: none;
        }
    }

    .user-content {
        display: flex;
        margin: 10px 20px;
    }
    .user-content-info {
        margin-left: 20px;
    }
`


export const Usuarios = () => {

   const dispatch = useDispatch()
   const state = useSelector(state => state.reducer)
   const { userList, userByName } = state
   const user = userByName.length > 0 ? userByName : userList

   console.log(userList)
   console.log(userByName)
    useEffect(() => {
        fetch('https://jsonplaceholder.typicode.com/users')
          .then(response => response.json())
          .then(json => dispatch({
              type: 'GET_DATA_USERS',
              payload: json
          }))
    }, [ dispatch ])


    const handleInputChange = (e) => {
        
        dispatch({
            type: 'GET_USER_BY_NAME',
            payload: e.target.value
        })

    }
  
    return (
        <UsuariosStyled>

            <input 
              type='search'
              placeholder='Busca por nombre'
              onChange={ handleInputChange }/>

            {
                user.map( ({ name, phone, username, website }) => {
                    return (
                        <div key={ name } className='user-content'>
                            <img src='https://www.w3schools.com/css/img_avatar.png' alt='usuario foto' width='150px'/>
                            <div className='user-content-info'>
                               <p><strong>Nombre:</strong> { name } </p>
                               <p><strong>Telefono:</strong> { phone } </p>
                               <p><strong>Nombre Usuario:</strong> { username } </p>
                               <p><strong>Sitio Web:</strong> { website } </p>
                            </div>
                        </div>
                    )
                })
            }

        </UsuariosStyled>
    )
}
