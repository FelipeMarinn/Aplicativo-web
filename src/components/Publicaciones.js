import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { Form } from './Form'
import styled from 'styled-components'
import { setData } from '../helpers/setData'

const PublicacionesStyled = styled.div` 
  max-width: 1200px;
  margin: 0 auto;
  margin-top: 100px;

  .posts { 
    display: grid;
    grid-row-gap: 2.3em;
    grid-auto-flow: columns;
    grid-column-gap: 50px;
    grid-template-columns: repeat(auto-fill, minMax(0, 300px));
    background: var(--background);
    justify-content: center;
    padding: 3em 0;
    margin: 0 20px;
  }
`

export const Publicaciones = () => {

    const dispatch = useDispatch()
    const { postList } = useSelector(state => state.postsReducer)

    useEffect(() => {

        setData('posts').then( data => dispatch({
          type: 'GET_DATA_POSTS',
          payload: data
        }))
    
    }, [dispatch])

    return (
        <PublicacionesStyled>

            <Form />

            <div className='posts' >
            {
                postList.map( ({ title, body }) => {
                    return (
                        <div key={ title }>
                            <h3>{ title }</h3>
                            <p>{ body }</p>
                        </div>
                    )
                })
            }
            </div>
            
        </PublicacionesStyled>
    )
}