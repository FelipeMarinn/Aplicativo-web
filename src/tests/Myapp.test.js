import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme';
import { Myapp } from '../components/Myapp';


it ('se renderiza correctamente', () => {

    const wrapper = shallow(
            <Myapp />
    )

    // expect(toJson(wrapper)).toMatchSnapshot()
    expect( wrapper ).toMatchSnapshot()
})