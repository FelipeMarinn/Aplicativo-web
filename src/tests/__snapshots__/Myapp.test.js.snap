// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`se renderiza correctamente 1`] = `
<div>
  <h1>
    Soy un componente
  </h1>
</div>
`;
